docs/src/description/constraints.md

```diff
@@ -2,8 +2,158 @@
 
 ## Boundary constraints (`boundary_constraints`)
 
-TODO
+The `boundary_constraints` section consists of expressions describing the expected value of columns in the main or auxiliary traces at the specified boundary. Column boundaries can be selected using boundary accessors. Valid boundary accessors are `.first`, which selects the first cell of the column to which it is applied, and `.last`, which selects the last cell of the column column to which it is applied.
+
+**Boundary constraints are optional.** It is equally valid to define an empty `boundary_constraints` section or to omit the `boundary_constraints` section entirely.
+
+Boundary constraints that are defined against auxiliary columns or that use random values from the built-in `$rand` array will be identified as auxiliary constraints.
```

Comment:
I'd say that boundary constraints which use `$rand` values must be defined against auxiliary trace columns (defining them against main trace columns should be an error).

Reply:
I've opened #46